service/rpc/server.go

```diff
@@ -60,16 +60,21 @@ func (s *ServerImpl) Run() error {
 		return err
 	}
 
+	rpcs := grpc.NewServer()
+	rpcs.Register(s.s)
+
 	go func() {
-		c, err := s.s.listener.Accept()
-		if err != nil {
-			panic(err)
-		}
 		defer s.s.listener.Close()
-
-		rpcs := grpc.NewServer()
-		rpcs.Register(s.s)
-		rpcs.ServeCodec(jsonrpc.NewServerCodec(c))
+		for {
+			c, err := s.s.listener.Accept()
```

Comment:
We need to signal this goroutine to bail when the server is told to stop, as it should not continue accepting connections. Maybe when the server is initialized (in `NewServer`) we create 2 channels, `stopChan` and `newConnChan`. In this loop we can use `select { ... }` and do the appropriate task depending on the channel.